lib/Sema/CSSimplify.cpp

```diff
@@ -3286,6 +3310,40 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       }
     }
   }
+  
+  // If we're about to fail lookup, but we are looking for members in a type
+  // that conforms to DynamicMemberLookupProtocol, then we resolve the reference
+  // to the subscript(dynamicMember:) member, and pass the member name as a
+  // string.
+  if (constraintKind == ConstraintKind::ValueMember && hasInstanceMembers &&
```

Comment:
Maybe split out the body of this if statement into a new function, since performMemberLookup() is already quite long

Reply:
performMemberLookup is long, but the logic is straight-forward.  I don't think that splitting it would make it easier to read or maintain, but it could always be done at some point independent of this patch.